docs/source/tree_examples.rst

```diff
@@ -0,0 +1,194 @@
+
+Examples with functions trees
+=============================
+
+It is often useful to structure code in a hierarchical way, e.g. to group related
+functions together. In ``dags``, this can be achieved by defining a so-called
+"functions tree".
+
+
+Functions from different namespaces
+-----------------------------------
+
+In large projects, function names can become lengthy when they share the same namespace.
+Using dags, we can concatenate functions from different namespaces.
+
+Suppose we define the following function in the module `linear.py`:
+
+.. code-block:: python
+
+    def f(x):
+        return 0.5 * x
+
+In another module, called `parabolic.py`, we define two more functions. Note,
+that there is a function `f` in this module as well.
+
+.. code-block:: python
+
+    def f(x):
+        return x**2
+
+    def h(f, linear__f):
+        return (f + linear__f) ** 2
+
+The function `h` takes two inputs:
+- `f` from `parabolic.py`, referenced directly as f within the current
+namespace.
+- `f` from `linear.py`, referenced using its namespace with a double
+underscore separator (`linear__f`).
```

Comment:
These bullet point are not rendered correctly in the docs.

Reply:
Hope the bullets will do. My rst skills have depreciated rapidly... Will convert into myst when we take a serious stab at it.